File: mobile/SmartSolarMobile/app/src/main/java/com/smartsolar/mobile/data/remote/dto/ReservationDashboardSummaryResponse.java
package com.smartsolar.mobile.data.remote.dto;

public class ReservationDashboardSummaryResponse {
    private long pendingReservations;
    private long approvedFutureReservations;
    private String generatedAtUtc;

    public ReservationDashboardSummaryResponse() { }

    public ReservationDashboardSummaryResponse(long pendingReservations, long approvedFutureReservations, String generatedAtUtc) {
        this.pendingReservations = pendingReservations;
        this.approvedFutureReservations = approvedFutureReservations;
        this.generatedAtUtc = generatedAtUtc;
    }

    public long getPendingReservations() { return pendingReservations; }
    public long getApprovedFutureReservations() { return approvedFutureReservations; }
    public String getGeneratedAtUtc() { return generatedAtUtc; }
}

File: src/SmartSolar.Application/DTOs/Reservations/ReservationDashboardSummaryResponse.cs
/*
 * File: ReservationDashboardSummaryResponse.cs
 * Project: Smart Solar Microgrid Trading System
 * Purpose: Exposes server-calculated reservation counts and their UTC comparison instant.
 * Note: Keep this header and update method-level comments as the code evolves.
 */
namespace SmartSolar.Application.DTOs.Reservations;

public sealed record ReservationDashboardSummaryResponse(
    long PendingReservations, long ApprovedFutureReservations, DateTime GeneratedAtUtc);
